src/istio/oscal-component.yaml

```diff
@@ -1,671 +1,1451 @@
+                exempt_namespaces_msg = sprintf("Exempted Namespaces: %s", [concat(", ", exempt_namespaces)])
+
+                validate {
+                  has_proxyv2_sidecar.result
+                }
+                msg = has_proxyv2_sidecar.msg
+
+                # Check for proxyv2 container in pod spec
+                no_proxyv2 = [sprintf("%s/%s", [pod.metadata.namespace, pod.metadata.name]) | pod := input.pods[_]; not contains_proxyv2(pod); not is_exempt(pod)]
+
+                has_proxyv2_sidecar = {"result": true, "msg": msg} {
+                  count(no_proxyv2) == 0
+                  msg := "All pods have Istio sidecar proxy."
+                } else = {"result": false, "msg": msg} {
+                  msg := sprintf("Istio sidecar proxy not found in pods: %s.", [concat(", ", no_proxyv2)])
+                }
+
+                contains_proxyv2(pod) {
+                  images := pod.spec.containers[_].image
+                  contains(images, "/proxyv2:")
+                }
+
+                is_exempt(pod) {
+                    pod.metadata.namespace in exempt_namespaces
+                }
+            type: opa
+        title: all-pods-istio-injected
+        uuid: 1761ac07-80dd-47d2-947e-09f67943b986
+      - description: |
+          domain:
+            kubernetes-spec:
+              create-resources: null
+              resources:
+              - description: ""
+                name: istioConfig
+                resource-rule:
+                  field:
+                    base64: false
+                    jsonpath: .data.mesh
+                    type: yaml
+                  group: ""
+                  name: istio
+                  namespaces:
+                  - istio-system
+                  resource: configmaps
+                  version: v1
+            type: kubernetes
+          lula-version: ""
+          metadata:
+            name: istio-metrics-logging-configured
+            uuid: 70d99754-2918-400c-ac9a-319f874fff90
+          provider:
+            opa-spec:
+              output:
+                observations:
+                - validate.msg
+                validation: validate.validate
+              rego: |
+                package validate
+
+                # Default policy result
+                default validate = false
+                default msg = "Not evaluated"
+
+                # Validate Istio configuration for metrics logging support
+                validate {
+                  check_metrics_enabled.result
+                }
+                msg = check_metrics_enabled.msg
+
+                check_metrics_enabled = { "result": true, "msg": msg } {
+                  input.istioConfig.enablePrometheusMerge
+                  msg := "Metrics logging supported"
+                } else = { "result": false, "msg": msg } {
+                  msg := "Metrics logging not supported"
+                }
```

Comment:
Can't tell if this already accounts for this, but Istio [enables this by default](https://istio.io/latest/docs/ops/integrations/prometheus/#option-1-metrics-merging) so "not set" or true should be acceptable here.

Reply:
Created an issue in our backlog to review.